UPGRADE-3.0.md

```diff
@@ -0,0 +1,110 @@
+
+# Upgrading from 2.x to 3.0
+
+API client release 3.0 contains few backward incompatible changes.
+
+This guide will help you upgrade your codebase.
+
+## `UserRecommendation::create()` now accepts only `$userId` and `$scenario` 
+`UserReccomentation::create()` accepts only two arguments: `$user_id` and `$scenario`.
+
+Both arguments are required. 
+
+All other arguments can be now set using new setters and are optional:
+- `setCount(int $count)`
+- `setRotationRate(float $rotationRate)`
+- `setRotationTime(int $rotationTime)`
+
+#### Before
+```php
+$recommendation = UserRecommendation::create('user-id', 5, 'scenario', 1.0, 3600);
+```
+
+#### After
+```php
+$recommendation = UserRecommendation::create('user-id', 'scenario')
+	->setCount(5)
+	->setRotationRate(1.0)
+	->setRotationTime(3600);
+```
+
+which is equivalent to
+
+```php
+$recommendation = UserRecommendation::create('user-id', 'scenario');
+$recommendation->setCount(5);
+$recommendation->setRotationRate(1.0);
+$recommendation->setRotationTime(3600);
+```
+
+## `Interaction` now accepts interaction type as parameter
+Matej now allows configuration of custom interaction types and interaction attributes.
+
+At the same time, it allows specifying interaction item using item alias instead
+of item ID. For that reason we removed static constructor methods for creating specific interaction types - `Interaction::detailView`, `Interaction::purchase`, `Interaction::bookmark` and `Interaction::rating`.
+
+We replaced them with constructors for creating Interaction based on `$itemId` or `$itemIdAlias`:
+
+```php
+Interaction::withItem(
+	string $interactionType,
+	string $userId,
+	string $itemId,
+	int $timestamp = null
+```
+
+```php
+Interaction::withAliasedItem(
```

Comment:
You don't find explanation of what's an interaction or merge in this repo either. I assumed this library is meant for users who are familiar with the Matej API.

Alias item is a way to provide custom ids by the client if they don't want to use item_id attribute. They can specify how their ids are called. This can be different for different item types on the page (they can send interaction for a job posting with one type of id and interactions for whole pages with different ids. To differentiate them, they have to make an alias. This will be described in the Matej documentation. @VincTheSecond is working on that).

I don't believe it should be explained in the migration tutorial.